clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -1012,6 +1013,28 @@ public void testChannelCloseWhileProcessingReceives() throws Exception {
         assertEquals(0, selector.completedReceives().size());
     }
 
+    /**
+     * Validate that correct subset of io metrics marked deprecated in docs
+     */
+    @Test
+    public void testIoMetricsHaveCorrectDoc() {
+        List<String> actual = asList("io-ratio", "io-wait-ratio");
+        List<String> deprecated = asList("iotime-total", "io-waittime-total");
+
+        // iterate through all metrics, since metrics.metric(metricName) requires 
+        // knowing metric group and tag set, which are not trivial dependencies
+        for (MetricName metricName : metrics.metrics().keySet()) {
```

Comment:
Could we make sure both `actual` and `deprecated` get verified? For example, this pass can pass if we use incorrect metrics name. i.e `if (actual.contains(name))` and `if (deprecated.contains(name))` return false.

Reply:
I'm not sure we should. Could you please elaborate more? Below are my thoughts.

What is expected to be verified by new suggested check?
1. Absence of human mistake during writing the test? Then it may become similar to testing the tests.
2. Actual presence of all these metrics in the registry? But I think this particular test shouldn't care if some of them are removed at some point. The goal here is "for these names, if they are registered, we expect such docs".

However, I'm concerning that if at some point **all of them** are removed, the test will stop doing anything meaningful without showing any sign of it.

Reply:
> However, I'm concerning that if at some point all of them are removed, the test will stop doing anything meaningful without showing any sign of it. But I don't see an elegant trigger to show it.

It seems to me making the test failed can remind developer - "oh, this test is failed and it unnecessary since I have removed those metrics from production".

Reply:
For example:
```java
        assertEquals(actual.size(), metrics.metrics().keySet().stream()
            .filter(m -> actual.contains(m.name()))
            .filter(m -> !m.description().toLowerCase(Locale.ROOT).contains("deprecated"))
            .count());

        assertEquals(deprecated.size(), metrics.metrics().keySet().stream()
            .filter(m -> deprecated.contains(m.name()))
            .filter(m -> m.description().toLowerCase(Locale.ROOT).contains("deprecated"))
            .count());
```

Reply:
That's the elegant solution I was looking for :)
Applied.